internal/rpc/digest_md5_integrity.go

```diff
@@ -0,0 +1,121 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"net"
+	"syscall"
+	"time"
+)
+
+type digestMD5IntegrityConn struct {
+	baseConnWrapper
+	buf bytes.Buffer
+
+	kic []byte
+	kis []byte
+
+	sendSeqNum int
+	readSeqNum int
+}
+
+func newDigestMD5IntegrityConn(conn net.Conn, kic, kis []byte) digestMD5Handler {
+	return &digestMD5IntegrityConn{
+		baseConnWrapper: baseConnWrapper{conn: conn},
+		kic:             kic,
+		kis:             kis,
+	}
+}
+
+func (d *digestMD5IntegrityConn) Write(b []byte) (n int, err error) {
+	if !d.writeDeadline.IsZero() && d.writeDeadline.Before(time.Now()) {
```

Comment:
I don't think this is necessary? The underlying call to `Write` should take care of that for us.